src/Simulation/Simulators.Tests/QuantumSimulatorTests/BasicTests.cs

```diff
@@ -20,12 +20,22 @@ public void QSimConstructor()
             {
                 Assert.Equal("Quantum Simulator", subject.Name);
             }
+            using (var subject = new SparseSimulator2())
+            {
+                Assert.Equal("SparseSimulator2", subject.Name);
+            }
         }
 
         [Fact]
         public void QSimVerifyPrimitivesCompleteness()
         {
-            using (var sim = new QuantumSimulator())
+            var simulators = new CommonNativeSimulator[] { 
```

Comment:
Similarly here, with `using` removed you may need to call `IDisposable.Dispose` from a `try`/`finally`.